added check to prevent checking filter rules against non-blockable URLs

snyderp opened the pull request:
Added `isBlockableProtocol` function in `ad_block_client.h/cc` and corresponding tests in `test/matching_test.cc`

Quick ignores URLs that don't start with one of the following (case insensitive, any of which can be prefix by `blob:`):
- http://
- https://
- ws://
- wss://

bbondy commented on ad_block_client.cc:
```diff
@@ -49,6 +49,52 @@ const int AdBlockClient::kFingerprintSize = 6;
+ *  - ws
+ *  - wss
+ */
+bool isBlockableProtocol(const char *url, int urlLen) {
```
I think we should add a mini state machine for this to avoid the allocations.  Similar to how the normal parsing works with keeping an enum for the current state and the current char that is being processed.

snyderp commented:
@bbondy Okie dokie, i've reimplemented the protocol check to use a state machine style flow.  It ended up being > 100 lines, so I moved it to its own file (`protocol.h/cc`), and changed the name of the test file to match (i.e. `test/protocol_test.cc`).

bbondy commented:
Thanks!
